packages/lesswrong/components/posts/PostsVideoCard.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useMemo, useRef } from "react";
+import { Components, registerComponent } from "../../lib/vulcan-lib";
+import { useHover } from "../common/withHover";
+import { AnalyticsContext } from "../../lib/analyticsEvents";
+import { cheerioParse } from "../../server/utils/htmlUtil";
+
+const styles = (theme: ThemeType) => ({
+  root: {
+    background: theme.palette.grey[0],
+    borderRadius: theme.borderRadius.default,
+    padding: 12,
+    marginBottom: 16,
+    color: theme.palette.text.slightlyIntense2,
+    fontFamily: theme.palette.fonts.sansSerifStack,
+    fontSize: 14,
+  },
+  header: {
+    display: "flex",
+    gap: "12px",
+    alignItems: "center",
+  },
+  karma: {
+    marginBottom: 10,
+  },
+  postTitle: {
+    overflow: "hidden",
+    display: "-webkit-box",
+    "-webkit-box-orient": "vertical",
+    "-webkit-line-clamp": 2,
+  },
+  metadata: {
+    marginTop: 4,
+    marginBottom: 12,
+    display: "flex",
+    alignItems: "center",
+    gap: "6px",
+  },
+  frame: {
+    border: "none",
+    width: "100%",
+    height: 183,
+  },
+});
+
+const videoHosts = [
+  "https://www.youtube.com",
+  "https://youtube.com",
+  "https://youtu.be",
+];
+
+const getEmbedAttribsFromHtml = (html: string): Record<string, unknown> | null => {
```

Comment:
Arguably it'd be better in the future to have a `videoUrl` field on the post object and do this parsing on the server-side when the post is created/edited, but this'll do for now.